lib/posix/pthread.c

```diff
@@ -61,18 +63,17 @@ static int pthread_concurrency;
 static K_MUTEX_DEFINE(pthread_once_lock);
 
 static const struct pthread_attr init_pthread_attrs = {
-	.priority = 0,
 	.stack = NULL,
+	.priority = 0,
 	.stacksize = 0,
-	.flags = PTHREAD_INIT_FLAGS,
-	.delayedstart = 0,
+	.initialized = true,
+	.cancellable = PTHREAD_CANCEL_ENABLE || true,
+	.detachstate = PTHREAD_CREATE_JOINABLE || true,
```

Comment:
are these supposed to be `&&` instead, otherwise they will always be `true`?

Reply:
They should always be true. The main issue is that one of the enable flags is conventionally defined as zero in many libcs. The enable flags are really just there symbolically.

Reply:
Fixed!